serverlessrepo/publish.py

```diff
@@ -174,6 +179,24 @@ def _is_conflict_exception(e):
     return error_code == 'ConflictException'
 
 
+def _wrap_s3_exception(e):
+    """
+    Wrap S3 access denied exception with a better error message.
+
+    :param e: boto3 exception
+    :type e: ClientError
+    :return: S3PermissionsRequired if S3 access denied or the original exception
+    """
+    error_code = e.response['Error']['Code']
+    message = e.response['Error']['Message']
+
+    if error_code == 'BadRequestException' and "Failed to copy S3 object" in message:
```

Comment:
This error message can happen in both access denied and object not found errors. Should either make the message work for both cases or should expand the check to be "Failed to copy S3 object. Access denied"